db/db_impl/db_impl_open.cc

```diff
@@ -143,11 +143,12 @@ DBOptions SanitizeOptions(const std::string& dbname, const DBOptions& src,
     result.wal_dir = result.wal_dir.substr(0, result.wal_dir.size() - 1);
   }
 
-  if (result.compaction_readahead_size == 0) {
-    if (result.use_direct_reads) {
+  if (result.compaction_readahead_size == Options().compaction_readahead_size ||
```

Comment:
There should be a constant, `kDefaultCompactionReadaheadSize`

Reply:
Not needed as the meta approach changes